src/components/SendNativeToken/sendNativeTokenUI.tsx

```diff
@@ -114,20 +117,26 @@ const SendNativeTokenUI = ({
 
   // Logic for estimating and sending transactions
   const estimateAndSend = async () => {
-    if (isValidAddress) {
-      // Estimate logic
-      const gasEstimation = await estimate();
-
-      if (!onlyEstimate) {
-        // Send logic
-        send();
-      }
-
-      if (debug) {
-        console.log('Transaction sent successfully');
+    try {
+      if (isValidAddress) {
+        // Estimate logic
+        const gasEstimation = await estimate();
+
+        if (!onlyEstimate) {
+          // Send logic
+          await send();
+        }
+
+        if (debug) {
+          console.log('Gas estimation:', gasEstimation);
+          console.log('Transaction sent successfully');
+        }
+      } else {
+        setError('Receiver address is not a valid blockchain address');
       }
-    } else {
-      setError('Receiver address is not a valid blockchain address');
+    } catch (error) {
+      // Handle errors from estimate or send methods
+      setError('Something wrong is happening to estimate and send transaction. Try again.');
```

Comment:
We can't give errors like this, where we're intentionally vague about what the error _could_ be

- It's either an estimation OR send error
- Explain properly what the user needs to do to "try again"
- Add a debug flag and a console.error which outputs the error
- The use of a wide try / catch block is making this difficult to differentiate which error this is
- Also add a link to the eEherspot documentation at the end of the console.error